SdFormatter.ino -> Universal 1MHz fallback

mikeysklar opened the pull request:
This adds a universal fast→1 MHz fallback for SD init and sectorCount() to improve compatibility with XTSD/Zetta and other timing-sensitive cards. Tested on Feather RP2040 and Feather ESP32 V2, with normal SD cards unaffected.

PR #42 attempted to do this, but was ESP32 specific. This removes any processor architecture and only slows down to 1MHz when necessary.

mikeysklar committed:
universal 1MHz fallback

This adds a universal fast→1 MHz fallback for SD init and sectorCount() to improve compatibility with XTSD/Zetta and other timing-sensitive cards. Tested on Feather RP2040 and Feather ESP32 V2, with normal SD cards unaffected.

hathach commented on examples/SdFormatter/SdFormatter.ino:
```diff
@@ -208,8 +206,27 @@ void setup() {
   if (!m_card || m_card->errorCode()) {
-    sdError("card init failed.");
-    return;
+    // Try slow 1 MHz fallback if fast init fails
```
can we just merge these 2 together as 

```C
  m_card = cardFactory.newCard(SD_CONFIG);
  if (!m_card || m_card->errorCode() || m_card->sectorCount() == 0) {
    // Try slow 1 MHz fallback if fast init fails
    SdSpiConfig slowCfg(SD_CS_PIN, SHARED_SPI, SD_SCK_MHZ(1));
    SdCard* slowCard = cardFactory.newCard(slowCfg);
    if (!slowCard || slowCard->errorCode() || slowCard->sectorCount() == 0) {
      sdError("card init failed.");
      return;
    }
    m_card = slowCard;
  }
```

mikeysklar committed:
Unify SD card initialization fallback

Combining error and zero-sector checks into a single 1 MHz retry path, tested and confirmed working on Feather RP2040 and Feather ESP32 V2.

mikeysklar commented:
Good call. 

Combined error and zero-sector checks into a single 1 MHz retry path, re-tested and confirmed working on both Feather RP2040 and Feather ESP32 V2.

hathach reviewed (approved):
perfect, thank you very much for the PR
